Handle empty array of ids when zoomTo/panTo

```
       Uncaught (in promise) AssertionError: Assertion failed. See https://openlayers.org/en/v6.15.1/doc/errors/#25 for details.
    at new AssertionError (AssertionError.js?15d4:38:1)
    at assert (asserts.js?c2e9:11:1)
    at View.fit (View.js?b093:1187:1)
    at origoiframeetuna.js:154:1
```

The map could stay where it is and not crash, or it could zoom to home somehow.

--- a/README.md
+++ b/README.md
@@ -8,6 +8,7 @@ Ger ett API för att styra delar av Origo inuti en `<iframe>` via https://develo
 - `maxZoom`: Maximal inzoomningsnivå. **Valfri**.
 - `zoomDuration`: Tid (i millisekunder) för att genomföra en zoomning. **Valfri**.
 - `allowedOrigins`: En array med strängar för tillåtna ursprung att acceptera meddelanden från. **Valfri**
+- `homeWhenZero`: En boolesk variabel för huruvida kartan ska panorera eller zooma till start-position när ett meddelande med kommandona 'zoomTo' eller 'panTo' inkommer med en tom 'ids'-array. Defaultvärde är `false`. **Valfri**
 
 ### Meddelanden
 

--- a/src/origoiframeetuna.js
+++ b/src/origoiframeetuna.js
@@ -94,18 +94,17 @@ const Origoiframeetuna = function Origoiframeetuna(options = {}) {
     );
   }
 
-  function getExtent(ids, targetLayer) {
+  async function getExtent(ids, targetLayer) {
     if (ids.length === 0) {
       if (!homeWhenZero) return null;
       return startExtent;
     }
 
-    return getFeatures(targetLayer, ids).then(featureArray => {
-      const coordinateArray = featureArray.map(feature =>
-        feature.getGeometry().getFirstCoordinate()
-      );
-      return boundingExtent(coordinateArray);
-    });
+    const featureArray = await getFeatures(targetLayer, ids);
+    const coordinateArray = featureArray.map(feature => feature.getGeometry().getFirstCoordinate());
+    const extent = boundingExtent(coordinateArray);
+
+    return extent;
   }
 
   return Origo.ui.Component({
@@ -148,24 +147,29 @@ const Origoiframeetuna = function Origoiframeetuna(options = {}) {
           applyFiltering(targetLayer);
         } else if (command === 'panTo') {
           // command to pan to an array of features. If they do not fit inside the view then they do not fit inside the view.
-          const targetExtent = getExtent(ids, targetLayer);
-          if (targetExtent === null) return;
-          viewer
-            .getMap()
-            .getView()
-            .setCenter(getCenter(targetExtent));
+          getExtent(ids, targetLayer).then(extent => {
+            if (!(extent === null)) {
+              viewer
+                .getMap()
+                .getView()
+                .setCenter(getCenter(extent));
+            }
+          });
         } else if (command === 'zoomTo') {
           // command to zoom to a an array of features
-          const targetExtent = getExtent(ids, targetLayer);
-          if (targetExtent === null) return;
-          viewer
-            .getMap()
-            .getView()
-            .fit(targetExtent, {
-              maxZoom,
-              duration: zoomDuration,
-              padding: [20, 20, 20, 20],
-            });
+
+          getExtent(ids, targetLayer).then(extent => {
+            if (!(extent === null)) {
+              viewer
+                .getMap()
+                .getView()
+                .fit(extent, {
+                  maxZoom,
+                  duration: zoomDuration,
+                  padding: [20, 20, 20, 20],
+                });
+            }
+          });
         } else {
           console.warn(
             'Received a message with an invalid command. Expected setFilter|setVisibleIDs|resetFilter|panTo|zoomTo.'

--- a/src/origoiframeetuna.js
+++ b/src/origoiframeetuna.js
@@ -6,9 +6,10 @@ import {boundingExtent, getCenter} from 'ol/extent';
  * @param {{ layerIDField: string, maxZoom: number=, zoomDuration: number= }} options
  */
 const Origoiframeetuna = function Origoiframeetuna(options = {}) {
-  const {layerIDField, maxZoom, zoomDuration, allowedOrigins} = options;
+  const {layerIDField, maxZoom, zoomDuration, allowedOrigins, homeWhenZero = false} = options;
 
   let viewer;
+  let startExtent;
 
   /** @type string|undefined */
   let cqlFilter;
@@ -93,6 +94,20 @@ const Origoiframeetuna = function Origoiframeetuna(options = {}) {
     );
   }
 
+  function getExtent(ids, targetLayer) {
+    if (ids.length === 0) {
+      if (!homeWhenZero) return null;
+      return startExtent;
+    } 
+
+    return getFeatures(targetLayer, ids).then(featureArray => {
+      const coordinateArray = featureArray.map(feature =>
+        feature.getGeometry().getFirstCoordinate()
+      );
+      return boundingExtent(coordinateArray);
+    })
+  }
+
   return Origo.ui.Component({
     name: 'origoiframeetuna',
     onInit() {
@@ -133,30 +148,26 @@ const Origoiframeetuna = function Origoiframeetuna(options = {}) {
           applyFiltering(targetLayer);
         } else if (command === 'panTo') {
           // command to pan to an array of features. If they do not fit inside the view then they do not fit inside the view.
-          getFeatures(targetLayer, ids).then(featureArray => {
-            const coordinateArray = featureArray.map(feature =>
-              feature.getGeometry().getFirstCoordinate()
-            );
+          const targetExtent = getExtent(ids, targetLayer)
+          if (targetExtent === null) return;
             viewer
               .getMap()
               .getView()
-              .setCenter(getCenter(boundingExtent(coordinateArray)));
-          });
+              .setCenter(getCenter(targetExtent))
+
         } else if (command === 'zoomTo') {
           // command to zoom to a an array of features
-          getFeatures(targetLayer, ids).then(featureArray => {
-            const coordinateArray = featureArray.map(feature =>
-              feature.getGeometry().getFirstCoordinate()
-            );
-            viewer
-              .getMap()
-              .getView()
-              .fit(boundingExtent(coordinateArray), {
-                maxZoom,
-                duration: zoomDuration,
-                padding: [20, 20, 20, 20],
-              });
-          });
+          const targetExtent = getExtent(ids, targetLayer)
+          if (targetExtent === null) return;
+          viewer
+            .getMap()
+            .getView()
+            .fit(targetExtent, {
+              maxZoom,
+              duration: zoomDuration,
+              padding: [20, 20, 20, 20],
+            });
+         
         } else {
           console.warn(
             'Received a message with an invalid command. Expected setFilter|setVisibleIDs|resetFilter|panTo|zoomTo.'
@@ -166,7 +177,9 @@ const Origoiframeetuna = function Origoiframeetuna(options = {}) {
     },
     onAdd(evt) {
       viewer = evt.target;
-
+      const map = viewer.getMap()
+      startExtent = map.getView().calculateExtent(map.getSize())
+      
       // in case the layer gets added _after_ we have received a message
       viewer
         .getMap()

--- a/src/origoiframeetuna.js
+++ b/src/origoiframeetuna.js
@@ -98,14 +98,14 @@ const Origoiframeetuna = function Origoiframeetuna(options = {}) {
     if (ids.length === 0) {
       if (!homeWhenZero) return null;
       return startExtent;
-    } 
+    }
 
     return getFeatures(targetLayer, ids).then(featureArray => {
       const coordinateArray = featureArray.map(feature =>
         feature.getGeometry().getFirstCoordinate()
       );
       return boundingExtent(coordinateArray);
-    })
+    });
   }
 
   return Origo.ui.Component({
@@ -148,16 +148,15 @@ const Origoiframeetuna = function Origoiframeetuna(options = {}) {
           applyFiltering(targetLayer);
         } else if (command === 'panTo') {
           // command to pan to an array of features. If they do not fit inside the view then they do not fit inside the view.
-          const targetExtent = getExtent(ids, targetLayer)
+          const targetExtent = getExtent(ids, targetLayer);
           if (targetExtent === null) return;
-            viewer
-              .getMap()
-              .getView()
-              .setCenter(getCenter(targetExtent))
-
+          viewer
+            .getMap()
+            .getView()
+            .setCenter(getCenter(targetExtent));
         } else if (command === 'zoomTo') {
           // command to zoom to a an array of features
-          const targetExtent = getExtent(ids, targetLayer)
+          const targetExtent = getExtent(ids, targetLayer);
           if (targetExtent === null) return;
           viewer
             .getMap()
@@ -167,7 +166,6 @@ const Origoiframeetuna = function Origoiframeetuna(options = {}) {
               duration: zoomDuration,
               padding: [20, 20, 20, 20],
             });
-         
         } else {
           console.warn(
             'Received a message with an invalid command. Expected setFilter|setVisibleIDs|resetFilter|panTo|zoomTo.'
@@ -177,9 +175,9 @@ const Origoiframeetuna = function Origoiframeetuna(options = {}) {
     },
     onAdd(evt) {
       viewer = evt.target;
-      const map = viewer.getMap()
-      startExtent = map.getView().calculateExtent(map.getSize())
-      
+      const map = viewer.getMap();
+      startExtent = map.getView().calculateExtent(map.getSize());
+
       // in case the layer gets added _after_ we have received a message
       viewer
         .getMap()

